core/server/services/members/middleware.js

```diff
@@ -69,6 +69,57 @@ const getOfferData = async function (req, res) {
     });
 };
 
+const getMemberNewsletters = async function (req, res) {
+    try {
+        const memberUuid = req.query.uuid;
+
+        const memberData = await membersService.api.members.get({
+            uuid: memberUuid
+        }, {
+            withRelated: ['newsletters']
+        });
+        if (!memberData) {
+            res.writeHead(400);
+            res.end('Email address not found.');
```

Comment:
Can we `return` here? If we add more code after this block in the future, the flow will leak through and we'll get the "already sent headers" error

Reply:
Done, thanks